### ISSUE TITLE
Scipy Factorial2 change

### ISSUE THREAD

**Issue by shivupa** [theochem/iodata#302]
SciPy's `factorial2` behavior changed in this commit https://github.com/scipy/scipy/commit/5e91097ea8307340de75dbd3730a794d08cb2bb1
(It seems like factorial2(x<1) should have always returned zero, but it used to only do that if `exact=True`)

When factorial2 is called [here](https://github.com/theochem/iodata/blob/ca6113b60fc967f3c280bd03374aaba1644d672e/iodata/overlap.py#L218) with 0 then the factorial array now gets `0` instead of `1`.

This causes the following issue with scipy 1.11.4
```
/usr/local/lib/python3.11/site-packages/iodata/overlap.py:277: RuntimeWarning: divide by zero encountered in scalar divide
  return np.sqrt((4.0 * alpha) ** sum(n) * (2.0 * alpha / np.pi) ** 1.5
  ```

**Comment by PaulWAyers** [comment]
This is addressed in pending pull requestion #301 . For now, the work-around is to use scipy version 1.10 or less.

**Comment by shivupa** [comment]
Yup sorry I missed that!

**Comment by FarnazH** [comment]
Thanks, @shivupa for raising an issue. I just merged PR #301, so the code should work with any version of `scipy`.

**Comment by shivupa** [comment]
Thanks!
